addons/mrp_account/wizard/mrp_wip_accounting.py

```diff
@@ -0,0 +1,150 @@
+                res['journal_id'] = default[1]
+        if 'reference' in fields_list:
+            res['reference'] = _("Manufacturing WIP - %(orders_list)s", orders_list=productions and format_list(self.env, productions.mapped('name')) or _("Manual Entry"))
+        if 'mo_ids' in fields_list:
+            res['mo_ids'] = [Command.set(productions.ids)]
+        return res
+
+    date = fields.Date("Date", default=fields.Datetime.now)
+    reversal_date = fields.Date(
+        "Reversal Date", compute="_compute_reversal_date", required=True,
+        store=True, readonly=False)
+    journal_id = fields.Many2one('account.journal', "Journal", required=True)
+    reference = fields.Char("Reference")
+    line_ids = fields.One2many(
+        'mrp.account.wip.accounting.line', 'wip_accounting_id', "WIP accounting lines",
+        compute="_compute_line_ids", store=True, readonly=False)
+    mo_ids = fields.Many2many('mrp.production')
+
+    def _get_overhead_account(self):
+        overhead_account = self.env.company.account_production_wip_overhead_account_id
+        if overhead_account:
+            return overhead_account.id
+        cop_acc = self.env['ir.property']._get_default_property('property_stock_account_production_cost_id', 'product.category')[1]
+        if cop_acc:
+            return cop_acc[1]
+        return self.env['ir.property']._get_default_property('property_stock_account_input_categ_id', 'product.category')[1]
+
+    def _get_line_vals(self, productions=False, date=False):
+        if not productions:
+            productions = self.env['mrp.production']
+        if not date:
+            date = datetime.now().replace(hour=23, minute=59, second=59)
+        compo_value = sum(
+            ml.quantity_product_uom * (ml.product_id.lot_valuated and ml.lot_id and ml.lot_id.standard_price or ml.product_id.standard_price)
+            for ml in productions.move_raw_ids.move_line_ids.filtered(lambda ml: ml.picked and ml.quantity and ml.date <= date)
+        )
+        overhead_value = productions.workorder_ids._cal_cost(date)
+        sval_acc = self.env['ir.property']._get_default_property('property_stock_valuation_account_id', 'product.category')[1]
+        sval_acc = sval_acc[1] if sval_acc else False
+        return [
+            Command.create({
+                'label': _("WIP - Component Value"),
+                'credit': compo_value,
+                'account_id': sval_acc,
+            }),
+            Command.create({
+                'label': _("WIP - Overhead"),
+                'credit': overhead_value,
+                'account_id': self._get_overhead_account(),
+            }),
+            Command.create({
+                'label': _("Manufacturing WIP - %(orders_list)s", orders_list=productions and format_list(self.env, productions.mapped('name')) or _("Manual Entry")),
+                'debit': compo_value + overhead_value,
+                'account_id': self.env.company.account_production_wip_account_id.id,
```

Comment:
what if that field isn't set? I think maybe we need some kind of check before opening the wizard that the necessary accounts are set in the settings.

Reply:
It's blank in that case, and if I understood correctly, this is ok since it should be auto-set by the CoA